text/0148-sampling-probability.md

```diff
@@ -0,0 +1,209 @@
+
+Applied correctly, this approach provides accurate estimates for
+population counts and distributions with support for ad-hoc queries
+over the data.
+
+### Changes proposed
+
+This proposal leads to three change requests that will be carried out in
+separate places in the OpenTelemetry specification.  These are:
+
+1. For tracing, the SpanData message type should be extended with 
+   the `sample_count` field defined above.
+2. For metrics aggregate data: Count information aggregated from
+   sample metric events will have floating point values in general.
+   Histogram and Counter data must be able to support floating point
+   values.
+3. For metrics raw events: Exemplars should be extended with the 
+   `sample_count` field defined above.
+
+### Example: Dapper tracing
+
+Google's [Dapper](https://research.google/pubs/pub36356/) tracing
+system describes the use of sampling to control the cost of trace
+collection at scale.
+
+### Example: Statsd metrics
+
+A Statsd counter event appears as a line of text, for example a metric
+named `name` is incremented by `increment` using a counter event (`c`)
+with the given `sample_rate`.
+
+```
+name:increment|c|@sample_rate
+```
+
+For example, a count of 100 that was selected for a 1-in-10 simple
+random sampling scheme will arrive as:
+
+```
+counter:100|c|@0.1
+```
+
+Probability 0.1 leads to a `sample_count` of 10.  Assuming the sample
+was selected using an unbiased algorithm (as by a "fair coin"), we can
+interpret this event as probabilistically equal to a count of `100/0.1
+= 1000`.
+
+## Internal details
+
+The statistical foundation of this technique is known as the
+Horvitz-Thompson estimator ("A Generalization of Sampling Without
+Replacement From a Finite Universe", JSTOR 1952).  The
+Horvitz-Thompson technique works with _unequal probability sampling_
+designs, enabling a variety of techniques for controlling properties
+of the sample.  
+
+For example, you can sample 100% of error events while sampling 1% of
+non-error events, and the interpretation of `sample_count` will be
+correct.
+
+### Bias, variance, and sampling errors
+
+There is a fundamental tradeoff between bias and variance in
+statistics.  The use of unbiased sampling leads unavoidably to
+increased variance.
+
+Estimating sampling errors and variance is out of scope for this
+proposal.  We are satisfied with the unbiased property, which guarantees
+that the expected value of totals derived from the sample equals the
+true population totals.  This means that statistics derived from
+sample data in this way are always accurate, and that more data will
+always improve precision.
+
+### Non-probabilistic rate-limiters
+
+Rate-limiting stages in a telemetry collection pipeline interfere with
+sampling schemes when they operate in non-probabilistic ways.  When
+implementing a non-probabilistic form of rate-limiting, processors
+MUST set the `sample_count` to zero.
```

Comment:
It would help to present a scenario here where _not_ setting `sample_count` to zero predicts the wrong population value. It's not immediately obvious to the reader why rate limiting invalidates the estimate.

Reply:
Added a section on the use of zero adjusted count.